### DISCUSSION TITLE
Solid quadrilateral is not drawing correclty

### DISCUSSION THREAD

**Discussion by johnthagen** [mozman/ezdxf#422] [Q&A] [answered]
Running `ezdxf` 0.16.2

```py
import ezdxf

doc = ezdxf.new("R2013")
msp = doc.modelspace()
msp.add_solid([(0.0, 0.0), (0.0, 10.0), (10.0, 10.0), (10.0, 0.0)])
doc.saveas("out.dxf")
```

I would expect this to render as a square, but when I open it in eDrawings Viewer or Autodesk DWG TrueView, it renders as an hourglass shape instead:

<img width="561" alt="Screen Shot 2021-04-29 at 3 41 41 PM" src="https://user-images.githubusercontent.com/10340167/116608605-689b4600-a901-11eb-991d-3e49b6ee67fc.png">

Am I doing something incorrectly? `Solid`'s don't seem to have the `close` parameter like `add_polyline2d` or `Hatch.paths.add_polyline_path`, that I've used before, so perhaps I'm misinterpreting how `Solid` is supposed to work?

**Comment by mozman** [comment]
Because ezdxf is an DXF interface, it writes the vertices as the DXF format stores the vertices, the last two vertices have to be reversed. This is a design decision done by Autodesk.

**Comment by mozman** [comment, marked as answer]
Added addtional documentation to SOLID and TRACE:

The SOLID entity stores the vertices in an unusual way, the last two vertices are reversed. The coordinates [(0, 0), (1, 0), (1, 1), (0, 1)] do not create a square as you would expect:

![solid0](https://user-images.githubusercontent.com/4230966/116646756-2b729a80-a979-11eb-9097-7600360671d1.png)

Reverse the last two vertices to get the `expected` square: [(0, 0), (1, 0), (0, 1), (1, 1)]

![solid1](https://user-images.githubusercontent.com/4230966/116646769-31687b80-a979-11eb-8957-4d2d671b3491.png)

**Reply by johnthagen** [reply]
The new docs are great, thank you!
